editoast/src/chartos/layers_description.rs

```diff
@@ -0,0 +1,49 @@
+use serde::{Deserialize, Serialize};
+use serde_yaml::{self};
+use std::fs;
+
+// select C.stuff from A inner join B C on C.id = C.id;
+//                       \___________________________/
+//                             a join expression
+//                            C is an alias for B
+type JoinExpr = String;
+
+pub const LAYERS_DESCRIPTION_PATH: &str = "./layers_description.yml";
+
+fn empty_vec() -> Vec<String> {
+    vec![]
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub struct View {
+    pub name: String,
+    pub on_field: String,
+    pub data_expr: String,
+    #[serde(default = "empty_vec")]
+    pub exclude_fields: Vec<String>,
+    #[serde(default = "empty_vec")]
+    pub joins: Vec<JoinExpr>,
+    pub cache_duration: u32,
+    #[serde(rename = "where", default = "empty_vec")]
+    pub where_expr: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub struct Layer {
+    pub name: String,
+    pub table_name: String,
+    pub views: Vec<View>,
+    #[serde(default)]
+    pub id_field: Option<String>,
+    #[serde(default)]
+    pub attribution: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub struct LayersDescription {
+    pub layers: Vec<Layer>,
+}
+
+pub fn parse_layers_description(file_path: &str) -> LayersDescription {
```

Comment:
- This function can be part of the implementation of `LayerDescription` and name `parse`.
- I don't think we need this parameter. The only use case could be to specify another config file.